File: frontend/widget/src/scenario-type.ts
import type { ScenarioType } from './types';

const SCENARIO_TYPES = new Set<ScenarioType>(['tooltip', 'modal', 'banner']);

export function normalizeScenarioType(value: unknown): ScenarioType {
  return typeof value === 'string' && SCENARIO_TYPES.has(value as ScenarioType)
    ? (value as ScenarioType)
    : 'tooltip';
}

File: backend/internal/models/scenario_type.go
package models

type ScenarioType string

const (
	ScenarioTypeTooltip ScenarioType = "tooltip"
	ScenarioTypeModal   ScenarioType = "modal"
	ScenarioTypeBanner  ScenarioType = "banner"
)

func NormalizeScenarioType(value string) (ScenarioType, bool) {
	scenarioType := ScenarioType(value)
	if scenarioType == "" {
		return ScenarioTypeTooltip, true
	}

	switch scenarioType {
	case ScenarioTypeTooltip, ScenarioTypeModal, ScenarioTypeBanner:
		return scenarioType, true
	default:
		return "", false
	}
}
